### ISSUE TITLE
Light Locker and GNOME 3.16

### ISSUE THREAD

**Issue by alucryd** [the-cavalry/light-locker#54]
It seems I can't use Light Locker together with GNOME 3.16 on Arch Linux, works fine with Openbox or XFCE. Here is what I get:

    light-locker --debug
    [gs_debug_init] gs-debug.c:106 (12:17:10):	 Debugging enabled
    [main] light-locker.c:126 (12:17:10):	 initializing light-locker 1.6.0
    [main] light-locker.c:128 (12:17:10):	 lock after screensaver 5
    [main] light-locker.c:129 (12:17:10):	 late locking 0
    [main] light-locker.c:130 (12:17:10):	 lock on suspend 1
    [init_session_id] gs-listener-dbus.c:1882 (12:17:10):	 Got session-id: /org/freedesktop/login1/session/c39
    [init_session_id] gs-listener-dbus.c:1887 (12:17:10):	 Got sd-session-id: c39
    [init_seat_path] gs-listener-dbus.c:1964 (12:17:10):	 Got seat: /org/freedesktop/DisplayManager/Seat0
    
    ** (light-locker:15889): WARNING **: screensaver already running in this session

I guess GNOME has its own screensaver running, but I can't find what it is and how to disable it. Have you got any ideas?

**Comment by ochosi** [comment]
Are you using Gnome with GDM? If so, light-locker won't work with that and you should use gnome-screensaver, which is now integrated with GDM.

If you are using LightDM, it should work fine, maybe you can tell us more about your setup.

**Comment by alucryd** [comment]
I am aware that gnome-screensaver is now part of GDM and did my tests using LightDM of course, It used to work fine with a GNOME session when I pushed light-locker in our repos (don't remember which GNOME we were at at the time). Today I realized it no longer worked with a Pantheon session (derivative of a GNOME session) launched via LightDM, and confirmed that vanilla GNOME launched via LightDM didn't work either.

**Comment by lots0logs** [comment]
I was able to get it to work by launching it from lightdm's xsession wrapper script just before gnome is launched:
`/etc/lightdm/Xsession`
```
....

/usr/bin/light-locker &
echo "X session wrapper complete, running session $@"

exec $@
```

**Comment by alucryd** [comment]
Nice find, I can't package lightdm with this workaround, but I'll write sth on the wiki when I have time.

**Comment by lots0logs** [comment]
I'm can't tell you how happy I am to FINALLY have a viable solution for locking the screen in GNOME without GDM! Kudos to @cavalier38 , @kelemeng  , and all contributors for making this possible :+1: :beers:

**Comment by CameronNemo** [comment]
Pantheon on Freya already uses light-locker, so you should not do that hack there.

**Comment by oblitum** [comment]
Not working for me too by default. I'm on a just fully updated Arch with Gnome 3 and LightDM (using a webkit greeter). I think I'll have to try @lots0logs's hack.

**Comment by oblitum** [comment]
Yes this hack works and made https://github.com/Antergos/gnome-shell-extension-status-menu-buttons functional.

**Comment by oblitum** [comment]
Actually I've changed the location for that line to an executable `.xsession` file in my home directory, since `/etc/lightdm/Xsession` loads it.

**Comment by postadelmaga** [comment]
Why this is closed, this problem is still present on gnome 3.22.x:
-  it is simply not possible to use light locker and gnome

**Comment by lots0logs** [comment]
@postadelmaga That's not true. It is certainly possible to do it. In fact our default GNOME installation uses light-locker.

**Comment by postadelmaga** [comment]
Here a related [thread](https://forum.antergos.com/topic/5473/gnome-not-locking-screen-automatically) if you can give some hint it will be useful.

**My experience:**
I have installed Antergos on 2 machine ( one now and one 6 month ago ) on both machine I have same issue, light locker is not working.

- _Settings  -> Power_: Blank screen (1 min)
- _Settings -> Privacy_: Screen Lock (yes)
- _Light locker_: 
  - Blank screen after (1 min)
  - Switch screen off after (2 min)

I have configured startup applications:
- Light Locker
- xset s 60 dpms 420 0 0

What is wrong in my settings ?
